rosidl_typesupport_freertps_cpp/resource/msg__type_support.cpp.template

```diff
@@ -29,39 +30,46 @@ extern "C"
 #include <rosidl_typesupport_freertps_cpp/visibility_control.h>
 
 @{header_file_name = get_header_filename_from_msg_name(spec.base_type.type)}@
-#include "@(spec.base_type.pkg_name)/@(subfolder)/@(header_file_name)__struct.hpp"
 #include "@(spec.base_type.pkg_name)/@(subfolder)/dds_freertps/@(header_file_name)__type_support.hpp"
-//#include "@(spec.base_type.pkg_name)/@(subfolder)/dds_freertps/ccpp_@(spec.base_type.type)_.h"
+#include "@(spec.base_type.pkg_name)/@(subfolder)/@(header_file_name)__struct.hpp"
 #include "rosidl_typesupport_freertps_cpp/identifier.hpp"
-#include <rosidl_typesupport_freertps_cpp/message_type_support.h>
+#include "rosidl_typesupport_freertps_cpp/message_type_support.h"
 
-// forward declaration of message dependencies and their conversion functions
+// forward declaration of conversion functions for message dependencies
 @[for field in spec.fields]@
 @[if not field.type.is_primitive_type()]@
-@[if field.type.pkg_name == 'builtin_interfaces']@
-#include "rosidl_typesupport_freertps_cpp/duration__type_support.hpp"
-#include "rosidl_typesupport_freertps_cpp/time__type_support.hpp"
-@[else]@
+@{
+dependency_subfolder = "msg"
+if field.type.type.endswith("Request") or field.type.type.endswith("Response"):
+    dependency_subfolder = "srv"
+}@
```

Comment:
Actually as the comment in the top of the file indicates a `subfolder` variable should already be available.